admin/views/Openedx_Woocommerce_Plugin_Settings.php

```diff
@@ -0,0 +1,266 @@
+     * Output the client ID settings field. 
+     *
+     * Retrieves the saved client ID value and outputs an input field and description text.
+     *
+     * @return void
+     */
+    public function openedx_client_id_callback()
+    {
+        $value = get_option( 'openedx-client-id' ); ?>
+
+        <input type="text" name="openedx-client-id" id="openedx-client-id" 
+            value="<?php echo esc_attr( $value ); ?>" required />
+
+        <p class="description">Identifier for OAuth application in your Open edX 
+            platform.</p>
+
+    <?php
+    }
+
+    /**
+     * Output the client secret settings field.
+     *
+     * Retrieves the saved client secret value and outputs a password input field and description text.
+     *
+     * @return void
+     */
+    public function openedx_client_secret_callback()
+    {
+
+        $value = get_option( 'openedx-client-secret' ); ?>
+
+        <input type="text" name="openedx-client-secret" id="openedx-client-secret" 
+            value="<?php echo esc_attr( $value ); ?>" required />
+
+        <p class="description">
+            Confidential key for OAuth application in your Open edX platform.
+        </p>
+
+    <?php
+    }
+
+    /** 
+     * Output the JWT token settings field.
+     *
+     * Retrieves the saved JWT token value and outputs a text input field, generate token button,
+     * and description text.
+     *
+     * @return void
+     */
+    public function openedx_jwt_token_callback()
+    {
+
+        $value = "12121adsad112ad";
+        $masked_value = str_repeat('*', strlen($value) - 4) . substr($value, -4); ?>
+
+        <div class="openedx-jwt-token-wrapper">
+
+            <input class="openedx-jwt-token-input" type="text" name="openedx-jwt-token" id="openedx-jwt-token"
+                value="<?php echo esc_attr($masked_value); ?>" disabled/>
+
+            <button class="button" type="button" id="generate-jwt-token">Generate JWT Token</button>
+
+        </div>
+
+        <p class="description"> Select the Generate Token button to obtain a JWT Token. </p>
+
+    <?php
+    }
+
+    /**
+     * Output introductory text for the settings section.
+     *
+     * Echoes text prompting the user to fill in and save the settings.
+     * 
+     * @return void
+     */
+    public function openedx_settings_section_callback()
+    {
+        printf( 'Configuring the necessary parameters here to establish
+        the connection between this plugin and your Open edX platform.');
+    }
+
+    /**
+     * Sanitize input string to only contain alphanumeric characters.
+     *
+     * This removes any non-alphanumeric characters from the given input string.
+     *
+     * @param string $input The input string to sanitize.
+     * @return string The sanitized input string containing only alphanumeric characters.
+     */
+    public function custom_sanitize_alphanumeric($input) 
+    {
+        return preg_replace('/[^a-zA-Z0-9]/', '', sanitize_text_field($input));
```

Comment:
At the time of making this function, I decided to use regular expressions because the "sanitize" function that comes by default from PHP and WordPress does not remove all the special characters so the alternative I had was to create that custom function for it. 

The reason why I didn't use JavaScript to watch the input and not allow it or use the HTML "pattern" property is that it's "easy" for someone with the knowledge to disable the JS that prohibits it or simply remove the "pattern" from the HTML input so if we have a separate function that takes care of this input cleaning it seems safer.

Reply:
Something I forgot to mention is that there is the possibility that instead of using regular expressions there is also the function sanitize_text_field() that can be used in that input but it is also processing when the request is sent and not when it is being written.

Reply:
I think we don't need to be so strict with this input; the important thing is not to allow vulnerabilities, particularly injections.

Reply:
Done, now the process is working with the native sanitizing WordPress function.

Reply:
Yes this will solve your problems